descriptive-link-5effbb.md

```diff
@@ -0,0 +1,205 @@
+
+This rule applies to any HTML or SVG element with the [semantic role](#semantic-role) of [link](https://www.w3.org/TR/wai-aria/#link) or a role that inherits from the link role, that is [included in the accessibility tree](#included-in-the-accessibility-tree) and has an [accessible-name](#accessible-name) that does not only consist of [whitespace](#whitespace).
+
+## Expectation
+
+The [visible](#visible) [text content](#text-content) of the target element, together with its [programmatically determined link context](#programmatically-determined-link-context) describe the purpose of the link.
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [2.4.4 Link Purpose (In Context)](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
+- [G91: Providing link text that describes the purpose of a link](https://www.w3.org/WAI/WCAG21/Techniques/general/G91)
+- [H30: Providing link text that describes the purpose of a link for anchor elements](https://www.w3.org/WAI/WCAG21/Techniques/html/H30)
+- [H24: Providing text alternatives for the area elements of image maps](https://www.w3.org/WAI/WCAG21/Techniques/html/H24)
+- [G53: Identifying the purpose of a link using link text combined with the text of the enclosing sentence](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G53)
+- [ARIA7: Using aria-labelledby for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA7)
+- [ARIA8: Using aria-label for link purpose](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA8)
+- [H77: Identifying the purpose of a link using link text combined with its enclosing list item](https://www.w3.org/WAI/WCAG21/Techniques/html/H77)
+- [H78: Identifying the purpose of a link using link text combined with its enclosing paragraph](https://www.w3.org/WAI/WCAG21/Techniques/html/H78)
+- [H79: Identifying the purpose of a link in a data table using the link text combined with its enclosing table cell and associated table header cells](https://www.w3.org/WAI/WCAG21/Techniques/html/H79)
+- [H81: Identifying the purpose of a link in a nested list using link text combined with the parent list item under which the list is nested](https://www.w3.org/WAI/WCAG21/Techniques/html/H81)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+Link text describes the purpose of the link
+
+```html
+<a href="#desc">See the description of this product</a> 
+
+<p id="desc">This product consists of several web pages</p>
+```
+
+#### Passed Example 2
+
+Accessible name describes the purpose of the link
+
+```html
+<a href="#main" aria-label="Go to the main content of the page"></a>
+
+<main>
+ <p id="main">This is the main content</p>
+</main>
+```
+
+#### Passed Example 3
+
+Link text together with its context describes the purpose of the link
+
+```html
+<p><a href="#desc">Click</a> to see the description of this product</p>
```

Comment:
Given that Understanding 2.4.4 explicitly recommends to put the link at the end of the sentence (after context) rather than at the start, I'd rather have the example showing the best practice.